release/release.go

```diff
@@ -111,36 +110,59 @@ var fileExists = func(path string) bool {
 	return err == nil
 }
 
-var filesystemRootType = func() (int64, error) {
-	var statfs syscall.Statfs_t
-	if err := syscall.Statfs("/", &statfs); err != nil {
-		return 0, fmt.Errorf("cannot statfs filesystem root")
+// filesystemRootType returns the filesystem type mounted at "/".
+var filesystemRootType = func() (string, error) {
+	// We scan /proc/mounts, which contains space-separated values:
+	// [irrelevant] [mount point] [fstype] [irrelevant...]
+	// Here are some examples on some platforms:
+	// WSL1       :  rootfs / wslfs rw,noatime 0 0
+	// WSL2       :  /dev/sdc / ext4 rw,relatime,discard,errors=remount-ro,data=ordered 0 0
+	// lxc on WSL2:  /dev/loop0 / btrfs rw,relatime,idmapped,space_cache,user_subvol_rm_allowed,subvolid=259,subvol=/containers/testlxd 0 0
+	// We search for mount point = "/", and return the fstype.
+	//
+	// This should be done by osutil.LoadMountInfo but that would cause a dependency cycle
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		return "", fmt.Errorf("cannot find root filesystem type: %v", err)
 	}
-	// Type is int32 on 386, use explicit conversion to keep the code
-	// working for both
-	return int64(statfs.Type), nil
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		data := strings.Fields(scanner.Text())
+		if len(data) < 3 || data[1] != "/" {
+			continue
+		}
+		return data[2], nil
+	}
+
+	if err = scanner.Err(); err != nil {
+		return "", fmt.Errorf("failed to find root filesystem type: %v", err)
+	}
+
+	return "", fmt.Errorf("failed to find root filesystem type: not in list")
```

Comment:
nitpick: Usually snapd's errors start with `cannot ...` like the first one in this function

```suggestion
		return "", fmt.Errorf("cannot find root filesystem type: %v", err)
	}

	return "", fmt.Errorf("cannot find root filesystem type: not in list")
```